### PR TITLE
Fetch and process information by using you tube api

### PR DESCRIPTION
**Pull request by Marjan1371**

### CODE REVIEW

**Comment by Marjan1371** [comment]
Added YouTube API integration and updated routes, services, and views

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -10,27 +10,48 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
+
+      # نصب جاوا 17
```
Comment (denzz312) [inline .github/workflows/build.yml:14]: Please only add English comments

File: app/controllers/YouTubeController.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+/**
+ * Controller to handle YouTube video search requests and display results.
+ */
+public class YouTubeController extends Controller {
+
+  public final YouTubeService youTubeService;
+  public final ExecutionContext ec; // تغییر به ExecutionContext
+
+  @Inject
+  public YouTubeController(YouTubeService youTubeService, ExecutionContext ec) { // تغییر
```
Comment (denzz312) [inline app/controllers/YouTubeController.java:23]: please only add English comments

File: build.sbt [hunk 3]
```diff
@@ -2,8 +2,42 @@ name := """TubeLytics"""
...
 
 scalaVersion := "2.13.15"
 
-libraryDependencies += guice
\ No newline at end of file
+libraryDependencies ++= Seq(
+  guice,
+  ws,
+  "org.playframework" %% "play-json" % "3.0.4",
+  "org.junit.jupiter" % "junit-jupiter-api" % "5.10.2" % Test,  // اضافه کردن junit-jupiter-api فقط برای تست
```
Comment (denzz312) [inline build.sbt:29]: please leave the comments in English

File: test/models/VideoTest.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        // بررسی عدم برابری دو ویدیو متفاوت
+        assertNotEquals(video1, video2);
+    }
+
+    @Test
+    public void testVideoHashCode() {
+        Video video1 = new Video("Sample Title", "Sample Description", "channelId123", "videoId123", "thumbnailUrl.jpg");
+        Video video2 = new Video("Sample Title", "Sample Description", "channelId123", "videoId123", "thumbnailUrl.jpg");
+
+        // بررسی برابر بودن hashCode برای ویدیوهای مشابه
```
Comment (denzz312) [inline test/models/VideoTest.java:54]: Same as previous comments

**Review by denzz312 — CHANGES_REQUESTED** [review]

**Comment by denzz312** [comment]
new pr is created https://github.com/SOEN6441-FALL2024/TubeLytics/pull/18

**Pull request closed without merging**